### PR TITLE
[design] "Since granularity"

### PR DESCRIPTION
**Pull request by umanwizard**
This is a lightweight design doc, as the feature being proposed is relatively small. Thus, I have deleted several sections from the doc. However, please feel free to ask if more details are needed.

[Rendered doc](https://github.com/umanwizard/materialize-1/blob/since_granularity/doc/developer/design/20230322_metrics_since_granularity.md)

### CODE REVIEW

**Comment by umanwizard** [comment]
Yes, that's true, we could do that. However, I thought it was nicer to be able to almost completely eliminate the chatter, rather than still having messages every 1 second for these tables.

**Comment by umanwizard** [comment]
@jkosh44 

> As far as I know we don't have an LD flag for the 30 day retention window

We do:
image

File: doc/developer/design/20230322_metrics_since_granularity.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# Explanation
+[explanation]: #explanation
+
+This design does not propose making this value user-configurable; we only want to tweak it for the "retained metrics relations", whose retention is currently set via LaunchDarkly to 30 days. We will make the following changes:
+
+1. Add a new LaunchDarkly parameter for `retained_metrics_since_granularity`, which will be set to 1 day by default.
+2. Add a "granularity" parameter to [`ReadPolicy::lag_writes_by`](https://github.com/MaterializeInc/materialize/blob/eda55b5c68/src/storage-client/src/controller.rs#L516-L530).
+3. For all objects with `is_retained_metrics_relation` set to true, use that LaunchDarkly parameter as the value of the parameter discussed in point 2.
+4. For all other objects, continue using 1 second as the default.
```
Comment (umanwizard) [inline doc/developer/design/20230322_metrics_since_granularity.md:24]:
```suggestion
4. For all other objects, continue using 1 second as the default.
```

**Comment by jkosh44** [comment]
> @jkosh44
> 
> > As far as I know we don't have an LD flag for the 30 day retention window
> 
> We do: image

Ah, well I stand corrected.

**Pull request merged**